src/fastq/collapse_reads_to_fragments/mod.rs

```diff
@@ -110,9 +110,17 @@ pub fn call_consensus_reads_from_paths(
 ) -> Result<(), Box<dyn Error>> {
     match fq3_out {
         None => {
-            info!("Reading input files:\n    {}\n    {}", fq1, fq2);
-            info!("Writing output to:\n    {}\n    {}", fq1_out, fq2_out);
-            match (fq1.ends_with(".gz"), fq2.ends_with(".gz"), fq1_out.ends_with(".gz"), fq2_out.ends_with(".gz")) {
+            info!(
+                "Reading input files:\n    {}\n    {}",
+                fq1.as_ref().display(),
+                fq2.as_ref().display()
+            );
+            info!(
+                "Writing output to:\n    {}\n    {}",
+                fq1_out.as_ref().display(),
+                fq2_out.as_ref().display()
+            );
+            match (fq1.as_ref().ends_with(".gz"), fq2.as_ref().ends_with(".gz"), fq1_out.as_ref().ends_with(".gz"), fq2_out.as_ref().ends_with(".gz")) {
```

Comment:
It should be possible to get rid of this whole match statement by either boxing the readers and writers or with conditionally initializing uninitialized variables. I'll post a suggestion "tomorrow".

Reply:
Does this really belong to this PR though? I mean we could easily put this in here but this should be somehow related to porting clap to structopt right?